playbooks/bootstrap-ssh.yml

```diff
@@ -0,0 +1,35 @@
+---
+- hosts: 127.0.0.1:all
+  gather_facts: false
+  tasks:
+    - name: set SSH key location
+      set_fact:
+        private_key: "{{ ansible_ssh_private_key_file | default(lookup('env','HOME') + '/.ssh/id_rsa') }}"
+
+- hosts: 127.0.0.1
```

Comment:
This might be something wonky with my machine since this used to work, but I'm getting an error with `bootstrap-ssh.yml`, it's skipping the local host:

```
$ ansible-playbook -l lab-gk8-worker4 -k playbooks/bootstrap-ssh.yml
SSH password:

PLAY [127.0.0.1:all] *******************************************************************************************************************************************************************************************

TASK [set SSH key location] ************************************************************************************************************************************************************************************
ok: [lab-gk8-worker4]

PLAY [127.0.0.1] ***********************************************************************************************************************************************************************************************
skipping: no hosts matched

PLAY [all] *****************************************************************************************************************************************************************************************************

TASK [Add SSH public key to ansible user user authorized keys] *************************************************************************************************************************************************
 [WARNING]: Unable to find '/home/dholt/.ssh/id_rsa.pub' in expected paths (use -vvvvv to see paths)

fatal: [lab-gk8-worker4]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'file'. Error was a <class 'ansible.errors.AnsibleError'>, original message: could not locate file in lookup: /home/dholt/.ssh/id_rsa.pub"}
        to retry, use: --limit @/local/dholt/deepops/playbooks/bootstrap-ssh.retry

PLAY RECAP *****************************************************************************************************************************************************************************************************
lab-gk8-worker4            : ok=1    changed=0    unreachable=0    failed=1

```

This solution seems to work:

```
$ git diff playbooks/bootstrap-ssh.yml
diff --git a/playbooks/bootstrap-ssh.yml b/playbooks/bootstrap-ssh.yml
index 1ed4caf..7400a6b 100644
--- a/playbooks/bootstrap-ssh.yml
+++ b/playbooks/bootstrap-ssh.yml
@@ -6,19 +6,20 @@
       set_fact:
         private_key: "{{ ansible_ssh_private_key_file | default(lookup('env','HOME') + '/.ssh/id_rsa') }}"

-- hosts: 127.0.0.1
-  connection: local
+- hosts: all
   gather_facts: false
   tasks:
     - name: Create SSH directory if it doesn't exist
       file:
         path: "{{ lookup('env','HOME') + '/.ssh' }}"
         state: directory
+      delegate_to: 127.0.0.1

     - name: Generate default SSH keypair if it doesn't exist
       command: ssh-keygen -q -t rsa -f "{{ private_key }}" -C "" -N ""
       args:
         creates: "{{ private_key }}"
+      delegate_to: 127.0.0.1

 - hosts: all
   become: true
```

Reply:
Good call. I was able to clean up the playbook to use a single block now.